internal/security/config/command/proxy/adduser/command.go

```diff
@@ -0,0 +1,340 @@
+	flagSet.StringVar(&cmd.jwtID, "id", "", "ID to use for linkage with JWT claim (usually the 'iss' field)")
+
+	flagSet.StringVar(&cmd.clientID, "client_id", "", "Optional manually-specified OAuth2 client_id.")
+	flagSet.StringVar(&cmd.clientSecret, "client_secret", "", "Optional manually-specified OAuth2 client_secret.")
+	flagSet.StringVar(&cmd.redirectUris, "redirect_uris", "", "OAuth2 redirect URL for browser-based users (required)")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+	if cmd.tokenType == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --token-type is required", os.Args[0])
+	}
+	if cmd.tokenType != interfaces.JwtTokenType && cmd.tokenType != interfaces.OAuth2TokenType {
+		return nil, fmt.Errorf("%s proxy adduser: argument --token-type must be either 'jwt' or 'oauth2'", os.Args[0])
+	}
+	if cmd.username == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --user is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.algorithm == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --algorithm is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.algorithm != "RS256" && cmd.algorithm != "ES256" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --algorithm must be either 'RS256' or 'ES256'", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.JwtTokenType && cmd.publicKeyPath == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --public_key is required", os.Args[0])
+	}
+	if cmd.tokenType == interfaces.OAuth2TokenType && cmd.redirectUris == "" {
+		return nil, fmt.Errorf("%s proxy adduser: argument --redirect_uris is required", os.Args[0])
+	}
+
+	return &cmd, err
+}
+
+func (c *cmd) Execute() (statusCode int, err error) {
+	switch c.tokenType {
+	case interfaces.JwtTokenType:
+		statusCode, err = c.ExecuteAddJwt()
+	case interfaces.OAuth2TokenType:
+		statusCode, err = c.ExecuteAddOAuth2()
+	default:
+		statusCode = interfaces.StatusCodeExitWithError
+		err = fmt.Errorf("unsupported token type %s", c.tokenType)
+	}
+
+	return
+}
+
+func (c *cmd) createConsumer() error {
+	// Create kong consumer with the specified username
+	// https://docs.konghq.com/hub/kong-inc/jwt/#create-a-consumer
+
+	form := url.Values{
+		"username": []string{c.username},
+	}
+	kongURL := strings.Join([]string{c.configuration.KongURL.GetProxyBaseURL(), "consumers"}, "/")
+	c.loggingClient.Info(fmt.Sprintf("creating consumer (user) on the endpoint of %s", kongURL))
+
+	formVal := form.Encode()
+	req, err := http.NewRequest(http.MethodPost, kongURL, strings.NewReader(formVal))
+	if err != nil {
+		return fmt.Errorf("Failed to prepare new consumer request %s: %w", c.username, err)
+	}
+	req.Header.Add(clients.ContentType, urlEncodedForm)
+	resp, err := c.client.Do(req)
+	if err != nil {
+		_ = resp.Body.Close()
```

Comment:
Why not use defer (here an elsewhere in the code)?

Reply:
Good catch. Made the implementation internally consistent.